tests/devices/i22/test_fswitch.py

```diff
@@ -0,0 +1,42 @@
+from typing import Any, Mapping
+from unittest import mock
+
+import pytest
+from ophyd_async.core import DeviceCollector, StandardReadable, set_sim_value
+
+from dodal.devices.i22.fswitch import FilterState, FSwitch
+
+
+@pytest.fixture
+async def fswitch() -> FSwitch:
+    async with DeviceCollector(sim=True):
+        fswitch = FSwitch("DEMO-FSWT-01:")
+
+    return fswitch
+
+
+async def assert_reading(
+    device: StandardReadable,
+    expected_reading: Mapping[str, Any],
+) -> None:
+    await device.stage()
+    reading = await device.read()
+    await device.unstage()
+    assert reading == expected_reading
+
+
+@mock.patch("time.time", mock.MagicMock(return_value=12345))
+async def test_reading_slits_reads_gaps_and_centres(fswitch: FSwitch):
+    set_sim_value(fswitch.filters.get(0), FilterState.OUT)
+    set_sim_value(fswitch.filters.get(1), FilterState.OUT)
+    set_sim_value(fswitch.filters.get(2), FilterState.OUT)
+
+    await assert_reading(
+        fswitch,
+        {
+            "num_filters": {
+                "timestamp": 12345,
```

Comment:
Can also do Any then don't have to patch time.time

Reply:
*`unittest.mock.ANY`, not `typing.Any`